src/admin/Guardian.sol

```diff
@@ -57,8 +54,8 @@ contract Guardian is IGuardian {
     }
 
     /// @inheritdoc IGuardian
-    function disputeMessageRecovery(bytes32 messageHash) external onlySafe {
-        gateway.disputeMessageRecovery(messageHash);
+    function disputeMessageRecovery(address adapter, bytes32 messageHash) external onlySafe {
```

Comment:
Looks like the only thing that is not tested yet, is that this method can only be called by `safe`. Do you want to add that as well?

Reply:
I instead added unit tests to generally test the `onlySafe` and `onlySafeAndOwner` modifiers